cirq-google/cirq_google/devices/coupler.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2024 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, Optional, Tuple, TYPE_CHECKING
+
+from cirq import ops, protocols
+
+if TYPE_CHECKING:
+    import cirq
+
+
+class Coupler(ops.Qid):
+    """A Qid representing the coupler between two qubits."""
+
+    _qubit1: ops.Qid
+    _qubit2: ops.Qid
+    _hash: Optional[int] = None
+    _comp_key: Optional[Tuple[Any, Any]] = None
+
+    def __init__(self, qubit1: ops.Qid, qubit2: ops.Qid) -> None:
+        """Creates a grid coupler between two qubits.
+
+        Note that the qubits will be implicitly sorted.
+        ie. `cirq_google.GridCoupler(q1, q2)` will be the same as
+        `cirq_google.GridCoupler(q2, q1)`.
+
+        Note that, if using custom Qid objects, the Qid must
+        have an ordering that allows for comparison.
+
+        Args:
+            qubit1: The first qubit/Qid of the pair
+            qubit2: The second qubit/Qid of the pair
+        """
+        if qubit1 < qubit2:
+            self._qubit1 = qubit1
+            self._qubit2 = qubit2
+        else:
+            self._qubit1 = qubit2
+            self._qubit2 = qubit1
+
+    def __hash__(self) -> int:
+        if self._hash is None:
+            self._hash = hash((self._qubit1, self._qubit2))
+        return self._hash
+
+    def _comparison_key(self):
+        if self._comp_key is None:
+            self._comp_key = (self._qubit1._comparison_key(), self.qubit2._comparison_key())
+        return self._comp_key
+
+    @property
+    def qubit1(self) -> ops.Qid:
+        return self._qubit1
+
+    @property
+    def qubit2(self) -> ops.Qid:
+        return self._qubit2
```

Comment:
I'd suggest adding a property to return a tuple of both qubits for easy decomposition (this would be an argument for zero-indexing since the tuple would be zero-indexed).
```python
@property
def qubits(self) -> tuple[ops.Qid, ops.Qid]:
    return (self._qubit0, self._qubit1)
```
Then can use this to decompose like:
```python
q0, q1 = coupler.qubits
```

Reply:
Added.